SDL/MiniDoublet.cuh

```diff
@@ -92,25 +90,503 @@ namespace SDL
+            {
+                // SDL::Hit upperHitMod(upperHit);
+                // upperHitMod.setXYZ(xn, yn, upperHit.z());
+                // setdeltaPhi_alpaka(lowerHit.deltaPhi_alpaka(upperHitMod));
+                shiftedX = xn;
+                shiftedY = yn;
+                shiftedZ = zUpper;
+                dPhi = deltaPhi_alpaka(acc, xLower, yLower, shiftedX, shiftedY);
+                noShiftedDphi = deltaPhi_alpaka(acc, xLower, yLower, xUpper, yUpper);
+            }
+            else
+            {
+                // SDL::Hit lowerHitMod(lowerHit);
+                // lowerHitMod.setXYZ(xn, yn, lowerHit.z());
+                // setdeltaPhi_alpaka(lowerHitMod.deltaPhi_alpaka(upperHit));
+                shiftedX = xn;
+                shiftedY = yn;
+                shiftedZ = zLower;
+                dPhi = deltaPhi_alpaka(acc, shiftedX, shiftedY, xUpper, yUpper);
+                noShiftedDphi = deltaPhi_alpaka(acc, xLower, yLower, xUpper, yUpper);
+            }
+        }
+        else
+        {
+            shiftedX = xn;
+            shiftedY = yn;
+            shiftedZ = zUpper;
+            dPhi = deltaPhi_alpaka(acc, xLower, yLower, xn, yn);
+            noShiftedDphi = deltaPhi_alpaka(acc, xLower, yLower, xUpper, yUpper);
+        }
+
+        // dz needs to change if it is a PS module where the strip hits are shifted in order to properly account for the case when a tilted module falls under "endcap logic"
+        // if it was an endcap it will have zero effect
+        if (modulesInGPU.moduleType[lowerModuleIndex] == SDL::PS)
+        {
+            dz = modulesInGPU.moduleLayerType[lowerModuleIndex] == SDL::Pixel ? zLower - zn : zUpper - zn; 
+        }
+
+        float miniCut = 0;
+        miniCut = modulesInGPU.moduleLayerType[lowerModuleIndex] == SDL::Pixel ?  dPhiThreshold(acc, rtLower, modulesInGPU, lowerModuleIndex,dPhi, dz) :  dPhiThreshold(acc, rtUpper, modulesInGPU, lowerModuleIndex, dPhi, dz);
+
+        pass = pass & (alpaka::math::abs(acc, dPhi) < miniCut);
+        if(not pass) return pass;
+
+        // Cut #4: Another cut on the dphi after some modification
+        // Ref to original code: https://github.com/slava77/cms-tkph2-ntuple/blob/184d2325147e6930030d3d1f780136bc2dd29ce6/doubletAnalysis.C#L3119-L3124
+
+        
+        float dzFrac = alpaka::math::abs(acc, dz) / alpaka::math::abs(acc, zLower);
+        dPhiChange = dPhi / dzFrac * (1.f + dzFrac);
+        noShiftedDphichange = noShiftedDphi / dzFrac * (1.f + dzFrac);
+        pass = pass & (alpaka::math::abs(acc, dPhiChange) < miniCut);
+
+        return pass;
+    }
 
     void printMD(struct miniDoublets& mdsInGPU, struct hits& hitsInGPU, SDL::modules& modulesInGPU, unsigned int mdIndex);
 
-    __global__ void createMiniDoubletsInGPUv2(struct SDL::modules& modulesInGPU, struct SDL::hits& hitsInGPU, struct SDL::miniDoublets& mdsInGPU, struct SDL::objectRanges& rangesInGPU);
+    struct createMiniDoubletsInGPUv2
```

Comment:
Good catch, I will update the code to make it consistent. The current Alpaka examples use struct instead of class, so I will use structs moving forward.

See here for example: https://github.com/alpaka-group/alpaka/blob/develop/example/helloWorld/src/helloWorld.cpp